FEAT - Implemented custom message parsers

StaelensRobin opened the pull request:
Hi, I implemented a way to inject custom Slack message parsers in the code based on a factory pattern.
This is the first time I do an open source PR so any feedback is welcome :)

Yitaek commented on Config.js:
```diff
@@ -32,7 +32,7 @@ async function decrypt(
 async function create() {
   require('dotenv').config()
```
Since you are setting dotenv config to envConfig, you could remove this. Also you can use this to catch errors in loading env variables if you want

Yitaek commented:
Hi @StaelensRobin , thanks for this PR! I've been meaning to add a way to allow custom slack msg parser and this is an elegant way to do it. My only ask is to update package.json with a minor version bump to indicate new feature and maybe add some blurb in the Readme about adding a custom parser like you did. You could even rename your file as ExampleParser.js or something to indicate that for others as well. 

Thanks again!

StaelensRobin commented:
Hi @Yitaek, no problem, very glad to help on this! 😄 
I change the message parsers' name to exampleMessageParser and added a section to the readme on how to add a new parser.

I also added a leaderboard feature for the amount of open PRs to the example parser.

Happy you liked this PR! 😄

Yitaek commented:
Great, thanks for the contributions @StaelensRobin
